quill-core/src/test/scala/io/getquill/PackageSpec.scala

```diff
@@ -50,6 +50,25 @@ class PackageSpec extends Spec {
     }
   }
 
+  "unquotes duble quotations" in {
+    val q: Quoted[EntityQuery[TestEntity]] = quote {
+      quote(query[TestEntity])
+    }
+    q.toString mustEqual "query[TestEntity]"
+  }
+
+  "unquotes quoted function bodies automatically" in {
+    implicit class QueryOps[Q <: Query[_]](q: Q) {
+      def allowFiltering = quote(infix"$q ALLOW FILTERING".as[Q])
+    }
+
+    val q: Quoted[Int => EntityQuery[TestEntity]] = quote {
```

Comment:
Overall LGTM

Why the explicit return type here? I just grabbed the branch and `PackageSpec` passes fine with:

```scala
val q = quote {
  (i: Int) =>
    query[TestEntity].allowFiltering
}
```
Is that just to explicitly show the inferred type?